kolibri_instant_schools_plugin/assets/src/views/sign-up-page/index.vue

```diff
@@ -75,22 +74,45 @@
         v-model="confirmedPassword"
       />
 
-      <ui-select
-        :name="$tr('selectFacility')"
-        :placeholder="$tr('selectFacility')"
-        :label="$tr('facility')"
-        :value="selectedFacility"
-        :options="facilityList"
-        :invalid="facilityIsInvalid"
-        :error="facilityIsInvalidText"
-        @blur="facilityBlurred = true"
-        @input="updateSelection"
-      />
+      <!-- ToS modal only shows up if the box empty prior to clicking -->
+      <!-- Using UI Checkbox for blur events and v-model -->
+      <label class="terms-agreement">
+        <button
+          type="button"
+          class="terms-agreement-view-prompt"
+          @click="showTerms = true"
+        >
+          {{ $tr('viewTermsOfServicePrompt') }}
+        </button>
+
+        <k-checkbox
+          :class="['terms-agreement-checkbox', termsNotAgreed ? 'invalid' : '']"
+          :checked="termsAgreed"
+          @change="termsAgreed = $event"
+          @blur="termsAgreementCheckboxBlurred = true"
```

Comment:
Still need to event this blur from k-checkbox.

Reply:
Whether or not that happens seems to be in your hands :open_hands:

Reply:
MERGED!